github/pulls.go

```diff
@@ -301,11 +301,11 @@ type pullRequestMergeRequest struct {
 func (s *PullRequestsService) Merge(owner string, repo string, number int, commitMessage string, options *PullRequestOptions) (*PullRequestMergeResult, *Response, error) {
 	u := fmt.Sprintf("repos/%v/%v/pulls/%d/merge", owner, repo, number)
 
-	pullRequestBody := &pullRequestMergeRequest{CommitMessage: &commitMessage}
+	pullRequestBody := pullRequestMergeRequest{CommitMessage: commitMessage}
```

Comment:
Hearing this kind of rationale for why we're using pointer is exactly why I did it, thanks @gmlewis. :)

At least I have a reason for why it's there now.

However, I'd be curious to put that claim to the test and benchmark. The struct contains 4 strings, and strings are quite inexpensive to copy (they're a [16-byte struct](https://godoc.org/reflect#StringHeader) on 64-bit systems). So if I'm not mistaken, it's a difference of copying a 4*16 = 64 byte struct on the stack, vs allocating a pointer on the heap. I don't know off the top of my head how the two would compare in performance and allocations, and I'd be curious to write a bench to learn more about that.

But, definitely not as part of this PR. I'll revert the change for this PR.